civitai-downloader-v2/src/ui/interactive.py

```
@@ -0,0 +1,617 @@
+        """Create progress tracking context manager."""
+        return ProgressContext(total, description, self)
+
+
+class LoadingAnimation:
+    """Loading animation context manager."""
+    
+    def __init__(self, message: str, interface: InteractiveInterface):
+        self.message = message
+        self.interface = interface
+        self.stop_animation = False
+        self.animation_thread: Optional[threading.Thread] = None
```

Ruff (0.12.2)

Undefined name `threading`

(F821)